CMakeLists.txt

```diff
@@ -1,11 +1,37 @@
-cmake_minimum_required(VERSION 3.21)
+cmake_minimum_required(VERSION 3.20)
 
 project(DartSharedLibrary VERSION 0.1)
 
+option(BUILD_SMAPLES "Build the Sampels" ON)
+
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 set(DART_DLL_DIR "${PROJECT_SOURCE_DIR}/src")
+if(NOT DART_DIR)
 set(DART_DIR "${PROJECT_SOURCE_DIR}/dart-sdk/sdk")
+endif()
+
+set( OUTPUT_PATH_EXT "x32" )
+if (CMAKE_SIZEOF_VOID_P EQUAL 8)
+    message(STATUS "Is 64 Bit")
+    set( IS_64_BIT 1 )
+    set( OUTPUT_PATH_EXT "x64" )
+else ()
+    message(STATUS "Is 32 Bit")
+endif ()
+
+set(OUTPUT_DEBUG "${OUTPUT_PATH_EXT}/Debug")
+set(OUTPUT_RELEASE "${OUTPUT_PATH_EXT}/Release")
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${PROJECT_SOURCE_DIR}/out/$<$<CONFIG:DEBUG>:${OUTPUT_DEBUG}>$<$<CONFIG:RELEASE>:${OUTPUT_RELEASE}>")
+set(EXECUTABLE_OUTPUT_PATH "${PROJECT_SOURCE_DIR}/out/${OUTPUT_PATH_EXT}/${CMAKE_BUILD_TYPE}")
+set(LIBRARY_OUTPUT_DIRECTORY "${PROJECT_SOURCE_DIR}/out/${OUTPUT_PATH_EXT}/${CMAKE_BUILD_TYPE}")
+
 
 add_subdirectory(src)
-add_subdirectory(examples)
\ No newline at end of file
+if(BUILD_SMAPLES)
+    add_subdirectory(examples)
+endif(BUILD_SMAPLES)    
+
+MESSAGE(STATUS "Build to ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}")
+MESSAGE(STATUS "Build to ${EXECUTABLE_OUTPUT_PATH}")
+MESSAGE(STATUS "Build to ${LIBRARY_OUTPUT_DIRECTORY}")
```

Comment:
If we want to have 3 messages here, it might be better to state what each one is doing. So for example:
```
"CMake output directory is ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}"
"Building executables to ${EXECUTABLE_OUTPUT_PATH}"
"Building libraries to ${LIBRARY_OUTPUT_DIRECTORY}"
```